### ISSUE TITLE
[Bug]: can not build in production mode, I'm using vue-3 and vite. It says:[commonjs--resolver] Failed to resolve entry for package "@tiptap/pm". The package may have incorrect main/module/exports specified in its package.json: Missing "." specifier in "@tiptap/pm" package

### ISSUE THREAD

**Issue by casatwy** [ueberdosis/tiptap#5202]
### Affected Packages

@tiptap/pm  vue-3 vite 

### Version(s)

"@tiptap/core": "^2.4.0"     "@tiptap/pm": "^2.4.0"

### Bug Description

when I run `vite build --base=/` to build my project, it failed and says

<img width="1427" alt="截屏2024-06-03 11 55 03" src="https://github.com/ueberdosis/tiptap/assets/1005781/b7849d43-fe9d-4a01-bb5d-9ba1be18200e">

```
error during build:
[commonjs--resolver] Failed to resolve entry for package "@tiptap/pm". The package may have incorrect main/module/exports specified in its package.json: Missing "." specifier in "@tiptap/pm" package
    at packageEntryFailure (file:///myproject/node_modules/vite/dist/node/chunks/dep-BKbDVx1T.js:49436:17)
    at resolvePackageEntry (file:///myproject/node_modules/vite/dist/node/chunks/dep-BKbDVx1T.js:49431:9)
    at tryNodeResolve (file:///myproject/node_modules/vite/dist/node/chunks/dep-BKbDVx1T.js:49203:20)
    at Object.resolveId (file:///myproject/node_modules/vite/dist/node/chunks/dep-BKbDVx1T.js:48953:28)
    at file:///myproject/node_modules/rollup/dist/es/shared/node-entry.js:19597:40
    at async PluginDriver.hookFirstAndGetPlugin (file:///myproject/node_modules/rollup/dist/es/shared/node-entry.js:19497:28)
    at async resolveId (file:///myproject/node_modules/rollup/dist/es/shared/node-entry.js:18178:26)
    at async ModuleLoader.resolveId (file:///myproject/node_modules/rollup/dist/es/shared/node-entry.js:18581:15)
    at async Object.resolveId (file:///myproject/node_modules/vite/dist/node/chunks/dep-BKbDVx1T.js:8258:10)
    at async PluginDriver.hookFirstAndGetPlugin (file:///myproject/node_modules/rollup/dist/es/shared/node-entry.js:19497:28)
```

the content of my package.json is:
```
{
  "name": "myproject",
  "private": true,
  "version": "0.0.0",
  "type": "module",
  "scripts": {
    "dev": "vite --host 127.0.0.1 --base=/",
    "build": "rm -rf ./dist && vite build --base=/",
    "preview": "vite preview"
  },
  "dependencies": {
    "@tiptap/core": "^2.4.0",
    "@tiptap/extension-bubble-menu": "^2.4.0",
    "@tiptap/extension-document": "^2.4.0",
    "@tiptap/extension-heading": "^2.4.0",
    "@tiptap/extension-paragraph": "^2.4.0",
    "@tiptap/extension-placeholder": "^2.4.0",
    "@tiptap/extension-text": "^2.4.0",
    "@tiptap/pm": "^2.4.0",
    "@tiptap/vue-3": "^2.4.0",
    "vue": "^3.4.27",
    "vue-router": "^4.3.2"
  },
  "devDependencies": {
    "@vitejs/plugin-vue": "^5.0.5",
    "sass": "^1.77.4",
    "typescript": "^5.4.5",
    "unplugin-auto-import": "^0.17.6",
    "unplugin-vue-components": "^0.27.0",
    "vite": "^5.2.12",
    "vite-plugin-compression": "^0.5.1",
    "vue-tsc": "^2.0.19"
  },
  "packageManager": "yarn@4.2.2"
}
```

### Browser Used

Chrome

### Expected Behavior

build successfully

### Dependency Updates

- [X] Yes, I've updated all my dependencies.

**Comment by nperez0111** [comment]
Did you import `@tiptap/pm` anywhere in your project? This package is mostly a re-export of prosemirror packages and normally would be imported at some subpath like: `@tiptap/pm/state`

This likely is that you are importing the top-level `@tiptap/pm` which does not exist on it's own

**Comment by casatwy** [comment]
> Did you import `@tiptap/pm` anywhere in your project? This package is mostly a re-export of prosemirror packages and normally would be imported at some subpath like: `@tiptap/pm/state`
> 
> This likely is that you are importing the top-level `@tiptap/pm` which does not exist on it's own

yes, I'm using `import { Node as pmNode } from "@tiptap/pm/model"` in my custom node, does it matter?

**Comment by casatwy** [comment]
previously I ran `yarn add @tiptap/pm`，so there is @tiptap/pm in my package.json (posted my package.json before.)
and then I ran `yarn remove @tiptap/pm`，it says `doesn't provide @tiptap/pm (p1f3fb), requested by @tiptap/extension-placeholder.`

like this:
<img width="1279" alt="截屏2024-06-03 17 05 16" src="https://github.com/ueberdosis/tiptap/assets/1005781/271ae2fe-acaf-43cf-b7fe-20cb3d4cee34">

I tried remove `node_modules` and run `yarn` again, it still appears😂Is there anything I can do?

**Comment by nperez0111** [comment]
> yes, I'm using import { Node as pmNode } from "@tiptap/pm/model" in my custom node, does it matter?

That should work


Hm, the peer dependencies warning seems a little suspect, perhaps a yarn upgrade to bump the peers to valid ranges?

**Comment by casatwy** [comment]
> > yes, I'm using import { Node as pmNode } from "@tiptap/pm/model" in my custom node, does it matter?
> 
> That should work
> 
> Hm, the peer dependencies warning seems a little suspect, perhaps a yarn upgrade to bump the peers to valid ranges?

<img width="453" alt="截屏2024-06-03 16 54 13" src="https://github.com/ueberdosis/tiptap/assets/1005781/e9f62c34-3201-44e9-8106-57f1473eca81">

the `package.json` of @tiptap/vue-3 has a "."，and I checked the package.json of @tiptap/pm, there is no "." like @tiptap/vue-3，and the error is `Missing "." specifier in "@tiptap/pm" package`. Maybe this is the problem?

here is the `package.json` of @tiptap/pm：
```json
{
  "name": "@tiptap/pm",
  "description": "prosemirror wrapper package for tiptap",
  "version": "2.4.0",
  "homepage": "https://tiptap.dev",
  "keywords": [
    "tiptap",
    "prosemirror"
  ],
  "license": "MIT",
  "funding": {
    "type": "github",
    "url": "https://github.com/sponsors/ueberdosis"
  },
  "type": "module",
  "exports": {
    "./changeset": {
      "types": "./changeset/dist/index.d.ts",
      "import": "./changeset/dist/index.js",
      "require": "./changeset/dist/index.cjs"
    },
    "./collab": {
      "types": "./collab/dist/index.d.ts",
      "import": "./collab/dist/index.js",
      "require": "./collab/dist/index.cjs"
    },
    "./commands": {
      "types": "./commands/dist/index.d.ts",
      "import": "./commands/dist/index.js",
      "require": "./commands/dist/index.cjs"
    },
    "./dropcursor": {
      "types": "./dropcursor/dist/index.d.ts",
      "import": "./dropcursor/dist/index.js",
      "require": "./dropcursor/dist/index.cjs"
    },
    "./gapcursor": {
      "types": "./gapcursor/dist/index.d.ts",
      "import": "./gapcursor/dist/index.js",
      "require": "./gapcursor/dist/index.cjs"
    },
    "./history": {
      "types": "./history/dist/index.d.ts",
      "import": "./history/dist/index.js",
      "require": "./history/dist/index.cjs"
    },
    "./inputrules": {
      "types": "./inputrules/dist/index.d.ts",
      "import": "./inputrules/dist/index.js",
      "require": "./inputrules/dist/index.cjs"
    },
    "./keymap": {
      "types": "./keymap/dist/index.d.ts",
      "import": "./keymap/dist/index.js",
      "require": "./keymap/dist/index.cjs"
    },
    "./markdown": {
      "types": "./markdown/dist/index.d.ts",
      "import": "./markdown/dist/index.js",
      "require": "./markdown/dist/index.cjs"
    },
    "./menu": {
      "types": "./menu/dist/index.d.ts",
      "import": "./menu/dist/index.js",
      "require": "./menu/dist/index.cjs"
    },
    "./model": {
      "types": "./model/dist/index.d.ts",
      "import": "./model/dist/index.js",
      "require": "./model/dist/index.cjs"
    },
    "./schema-basic": {
      "types": "./schema-basic/dist/index.d.ts",
      "import": "./schema-basic/dist/index.js",
      "require": "./schema-basic/dist/index.cjs"
    },
    "./schema-list": {
      "types": "./schema-list/dist/index.d.ts",
      "import": "./schema-list/dist/index.js",
      "require": "./schema-list/dist/index.cjs"
    },
    "./state": {
      "types": "./state/dist/index.d.ts",
      "import": "./state/dist/index.js",
      "require": "./state/dist/index.cjs"
    },
    "./tables": {
      "types": "./tables/dist/index.d.ts",
      "import": "./tables/dist/index.js",
      "require": "./tables/dist/index.cjs"
    },
    "./trailing-node": {
      "types": "./trailing-node/dist/index.d.ts",
      "import": "./trailing-node/dist/index.js",
      "require": "./trailing-node/dist/index.cjs"
    },
    "./transform": {
      "types": "./transform/dist/index.d.ts",
      "import": "./transform/dist/index.js",
      "require": "./transform/dist/index.cjs"
    },
    "./view": {
      "types": "./view/dist/index.d.ts",
      "import": "./view/dist/index.js",
      "require": "./view/dist/index.cjs"
    }
  },
  "files": [
    "changeset/**",
    "collab/**",
    "commands/**",
    "dropcursor/**",
    "gapcursor/**",
    "history/**",
    "inputrules/**",
    "keymap/**",
    "markdown/**",
    "menu/**",
    "model/**",
    "schema-basic/**",
    "schema-list/**",
    "state/**",
    "tables/**",
    "trailing-node/**",
    "transform/**",
    "view/**"
  ],
  "dependencies": {
    "prosemirror-changeset": "^2.2.1",
    "prosemirror-collab": "^1.3.1",
    "prosemirror-commands": "^1.5.2",
    "prosemirror-dropcursor": "^1.8.1",
    "prosemirror-gapcursor": "^1.3.2",
    "prosemirror-history": "^1.3.2",
    "prosemirror-inputrules": "^1.3.0",
    "prosemirror-keymap": "^1.2.2",
    "prosemirror-markdown": "^1.12.0",
    "prosemirror-menu": "^1.2.4",
    "prosemirror-model": "^1.19.4",
    "prosemirror-schema-basic": "^1.2.2",
    "prosemirror-schema-list": "^1.3.0",
    "prosemirror-state": "^1.4.3",
    "prosemirror-tables": "^1.3.5",
    "prosemirror-trailing-node": "^2.0.7",
    "prosemirror-transform": "^1.8.0",
    "prosemirror-view": "^1.32.7"
  },
  "repository": {
    "type": "git",
    "url": "https://github.com/ueberdosis/tiptap",
    "directory": "packages/pm"
  },
  "scripts": {
    "build": "npm run build:changeset && npm run build:collab && npm run build:commands && npm run build:dropcursor && npm run build:gapcursor && npm run build:history && npm run build:inputrules && npm run build:keymap && npm run build:markdown && npm run build:menu && npm run build:model && npm run build:schema-basic && npm run build:schema-list && npm run build:state && npm run build:tables && npm run build:trailing-node && npm run build:transform && npm run build:view",
    "build:changeset": "tsup \"changeset/index.ts\" --out-dir changeset/dist",
    "build:collab": "tsup \"collab/index.ts\" --out-dir collab/dist",
    "build:commands": "tsup \"commands/index.ts\" --out-dir commands/dist",
    "build:dropcursor": "tsup \"dropcursor/index.ts\" --out-dir dropcursor/dist",
    "build:gapcursor": "tsup \"gapcursor/index.ts\" --out-dir gapcursor/dist",
    "build:history": "tsup \"history/index.ts\" --out-dir history/dist",
    "build:inputrules": "tsup \"inputrules/index.ts\" --out-dir inputrules/dist",
    "build:keymap": "tsup \"keymap/index.ts\" --out-dir keymap/dist",
    "build:markdown": "tsup \"markdown/index.ts\" --out-dir markdown/dist",
    "build:menu": "tsup \"menu/index.ts\" --out-dir menu/dist",
    "build:model": "tsup \"model/index.ts\" --out-dir model/dist",
    "build:schema-basic": "tsup \"schema-basic/index.ts\" --out-dir schema-basic/dist",
    "build:schema-list": "tsup \"schema-list/index.ts\" --out-dir schema-list/dist",
    "build:state": "tsup \"state/index.ts\" --out-dir state/dist",
    "build:tables": "tsup \"tables/index.ts\" --out-dir tables/dist",
    "build:trailing-node": "tsup \"trailing-node/index.ts\" --out-dir trailing-node/dist",
    "build:transform": "tsup \"transform/index.ts\" --out-dir transform/dist",
    "build:view": "tsup \"view/index.ts\" --out-dir view/dist"
  }
}
```

**Comment by nperez0111** [comment]
Right, but it has no meaningful export for `.` which is why it is not there. It should not need one if I'm not mistaken

**Comment by nperez0111** [comment]
Also strange to me that it is being interpretted as commonjs when it is an ES module?

> [commonjs--resolver] Failed to resolve entry for package "@tiptap/pm"

**Comment by casatwy** [comment]
@nperez0111 problem solved!

I'm using `build.rollupOptions.output.manualChunks` in `vite.config.ts`.
After delete these options, it works fine!

Thank you for your help!

<img width="386" alt="截屏2024-06-03 18 22 31" src="https://github.com/ueberdosis/tiptap/assets/1005781/b864a4b6-f87c-4b80-bbcf-d453d5a4ab26">

**Comment by jvdl** [comment]
For anyone else that might land here, I ran into this issue as well and I can't omit using manual chunks in my build so commenting/removing it wasn't an option. 

In my specific scenario I generate a chunkmap based on a set of matches to create chunks, e.g. the config passed to my mapping function looks like:

```
{
  react: ["react"],
  "react-libs": [
    "react-dom",
    "react-router",
    ...
  ],
  vendor:  [],
}
```

I added an empty chunk for "vendor" that all non-explicitly mapped items go into, and so that is why Tiptap ended up in my manual chunk config.

I found that _explicitly omitting_ tiptap from this manual chunk map will avoid this error. I haven't figured out _why_ yet and will update this comment/thread if I do ever find out, but I suspect that either Rollup does something funky because of the chunk map, or dependency ordering is relevant here.
